6th PR: Added Profile Settings Page and connect TutorView and StudentView with Navbar

FrankFurter123 opened the pull request:
### Added Profile Settings Page and connect TutorView and StudentView with Navbar

---

## Overview  
### What does this PR do?  
This PR implements a **Settings page** divided into four panels:  
- Public Profile  
- Account  
- Emails  
- Password & Security  

It also connects the `TutorView.jsx` page to the shared `StudentNavbar.jsx` to unify the experience for both student and tutor roles. In addition, dark theme support is now available on the Settings and TutorView pages.

---

## Key Features & Changes  
### What has been added or changed?  
✔ **Core Feature Implementation**  
- Created a full Settings page with form fields and save functionality placeholders  
- Added four tabs (Public Profile, Account, Emails, Password)  

✔ **UI/UX Enhancements**  
- Matching layout to GitHub-style settings interface  
- Light/Dark mode toggling support for all sections  
- Consistent spacing, hierarchy, and helper text

✔ **Routing & Navbar Enhancements**  
- Connected `/student/view` and `/tutor/view` to shared `StudentNavbar.jsx`  
- Ensured dark mode is consistent on `TutorView.jsx`  
- Fixed broken link logic in `App.jsx` to include routes wrapped with `StudentLayout`

✔ **Theme Support**  
- Extended dark mode support to all newly added components  

---

## Why This Is Needed  
### What problem does this solve?  
Students and tutors need a **central place to manage their profile**, email, password, and personal account details. This PR introduces that settings functionality and aligns the tutor view with the global student-tutor layout pattern.

---

## Related Issue  
This PR addresses issue #141 

---

## Implementation Details  
- Used state hooks and panel switching logic inside `Settings.jsx`  
- Dark mode inherited from global state passed via props  
- Connected navigation routes in `App.jsx`  
- Removed outdated styles from `TutorView.css`

---

## UML DIagram
![settings_page_uml](https://github.com/user-attachments/assets/b831930f-2fc1-4fca-885c-9edf505998a2)


---

## How to Test This PR  
1. Login as a student or tutor  
2. Navigate to `/student/view` or `/tutor/view`  
3. Open the profile dropdown in the top right and click **Settings**  or you can also click  **Settings**  on the right panel.
4. Verify each panel loads:
   - Public Profile: Avatar + full name, email, bio  
   - Account: Username, location, occupation, Discord, GitHub  
   - Emails: Primary + backup emails  
   - Password & Security: Password change inputs  
5. Toggle **dark theme** and confirm changes across all settings sections  
6. Visit Tutor View and ensure it uses the same global navbar and dark theme  

---

## Files Added/Modified  

| File Path                        | Description                            |
|----------------------------------|----------------------------------------|
| `frontend/src/Components/Settings.jsx`      | Main settings component with 4 sections |
| `frontend/src/Styles/Settings.css`          | Styles for GitHub-style layout          |
| `frontend/src/Components/TutorView.jsx`     | Now uses global StudentNavbar           |
| `frontend/src/Styles/TutorView.css`         | Cleaned up obsolete styles              |
| `frontend/src/Components/StudentNavbar.jsx` | Updated to support tutor role globally  |
| `frontend/src/Components/StudentLayout.jsx` | Now wraps tutor and student views       |
| `frontend/src/App.jsx`                      | Routes set for settings + TutorView     |

---

## Screenshots

**Public Profile Panel**  
<img width="1079" alt="Screenshot 2025-04-13 at 00 04 17" src="https://github.com/user-attachments/assets/d8b2827a-966a-4fa7-9e45-f5f449d21d65" />

**Account Settings Panel**  
<img width="1076" alt="Screenshot 2025-04-13 at 00 05 10" src="https://github.com/user-attachments/assets/1e65b73b-0653-4e5d-bd71-5cccfb891ab8" />

**Email Settings Panel**  
<img width="1073" alt="Screenshot 2025-04-13 at 00 05 28" src="https://github.com/user-attachments/assets/513d1889-67cd-4619-8811-17a532c3f341" />

**Password & Security Panel**  
<img width="1082" alt="Screenshot 2025-04-13 at 00 05 38" src="https://github.com/user-attachments/assets/797261ef-5c4d-4aa4-b85c-12e579991365" />

---

## AI Code Usage  
- [x] **No**  
---

## Checklist for Reviewers  
- [ ] Settings page opens and all panels load correctly  
- [ ] Dark mode applies consistently  
- [ ] Navbar links work for both student and tutor  
- [ ] TutorView uses shared layout and navigation  
- [ ] Code style and routing logic follows project conventions  

---

## Next Steps & Future Enhancements  
- Connect these settings to backend (update user info, upload avatar, etc.)  for my next PR
- Add error handling and form validation  
- Persist theme state across sessions  
- Load current user’s data dynamically

---

Thanks for reviewing! Let me know if you want me make adjustments before merge! Because we are aiming for MVP, I can still add more things that may not be fully functional but will make the design look better. 

– Franklin

AbdulAlharbi commented:
### Peer Review : Great Job on This PR!

Hey Frank! — just finished reviewing your PR titled **"Added Profile Settings Page and connect TutorView and StudentView with Navbar"**. Here’s my feedback:

---

## What Looks Great

- The **Settings page** is cleanly built, thoughtfully structured, and mirrors familiar UX patterns (like GitHub) — love it!
- The **four-tab layout** (Public Profile, Account, Emails, Password & Security) makes it easy to navigate and scale in the future.
- Unifying the **TutorView** and **StudentView** under a shared navbar is a smart design choice — improves maintainability and consistency.
- Great job ensuring **dark mode support** across all components — it works seamlessly.
- Your PR description is **thorough and reviewer-friendly**, including:
  - A clear summary of what changed and why.
  - UML diagram ✅
  - Screenshots ✅
  - Test instructions ✅
  - Related issue (#141) is properly linked ✅

---

## Suggestions for Improvement

> These are minor UX improvements that can be made now or in your follow-up backend PR:

- **Add a character limit to the bio field** (e.g., 280 or 500 characters). This will improve layout control and user clarity. Bonus points for adding a live counter!
- Include a **“Cancel” or “Discard Changes”** button beside “Save” in each section. It adds realism and improves UX by letting users back out without making changes.
- In the **Public Profile** panel, under the profile picture, consider changing the “Edit” button label to something more descriptive like `Change Photo` or `Upload Avatar` — "Edit" appears often and could use differentiation.
- For added polish: show a **success message** or disable the “Save” button after saving, so users know the action was registered.
- (Future feature) Consider persisting **dark mode preference** using `localStorage` — currently resets on refresh.

---

## Tested This Feature

I followed your test plan and verified the following:
- ✅ All four settings panels load and display the correct fields.
- ✅ Dark mode toggles and applies correctly across pages.
- ✅ Navbar routes work for both student and tutor views.
- ✅ TutorView uses the updated global layout with shared navigation.
- ✅ Settings styling is responsive and readable.

Everything works as expected — great job and thank you for this! 

---

## Next Steps

- You mentioned you're connecting the settings to the backend in your next PR — awesome! That’ll complete the functionality.
- You could explore using a form validation library (e.g., React Hook Form + Zod/Yup) to handle dynamic user input and error feedback.
- Let us know if you’d like help testing avatar uploads or form reset logic later!

---
## Questions? 
- How are you planning to load the current user’s data into each panel?
- How do you envision avatar upload working later?
- Do you think we should persist dark mode choice across sessions?
---
### Final Thoughts

This is a **meaningful and well-executed PR** that pushes the MVP forward in a big way. It’s clean, forward-compatible, and shows good product thinking. Really nice work, Franklin — let me know if you want to co-work on polishing the save interactions or backend integration!

Looking forward to seeing the next version live!   
— *Abdulrahman*

FrankFurter123 commented:
Hey @AbdulAlharbi , thank you so much for your thoughtful review and detailed feedback — really appreciate the time and care you put into testing and writing it up!

---

### ✅ Implemented All Suggestions

Your UX improvement ideas were what I needed from this Peer review — I’ve gone ahead and added all of them:

-  **Bio Character Limit**: Set a 500-character max with a live counter to help users stay within the limit.
-  **Discard Changes Button**: Added a “Discard Changes” button next to “Save” in each section for better usability.
-  **Change Avatar Button**: Replaced “Edit” with “Change Picture” in the Public Profile panel for clarity.
-  **Save Confirmation**: After saving, a success message appears and the Save button becomes disabled to signal the change was applied.

---

###  Answers to Your Questions

> **How are you planning to load the current user’s data into each panel?**  
Dorian from the backend team has already built the logic for data transmission to the Settings section and Profile Dashboard. I’m working closely with him to hook everything up and make sure no fields are missed.

> **How do you envision avatar upload working later?**  
The avatar will be collected during login and passed to this settings panel, where users can update it anytime. It will also reflect on their profile dashboard automatically.

> **Do you think we should persist dark mode choice across sessions?**  
100% agree. Right now, I’ve been adding dark theme styles manually per component to keep things consistent before the presentation. If we have time before the final MVP demo (just 2 weeks away!), I’d love to implement a persistent `localStorage` solution for this.

---

Thanks again.

— Frank

AbdulAlharbi commented:
Hey @FrankFurter123 ! 
Thanks so much for the fast turnaround, Frank — this is 🔥

---

### ✅ Everything Looks Amazing!

- The **bio limit with live counter** is such a nice touch — super intuitive and polished.
- Adding a **“Discard Changes”** button across sections really elevates the UX. That’s one of those small things that makes a big difference!
- Updating “Edit” to **“Change Picture”** looks much clearer now — great call on that one.
- Love that you included a **confirmation message and Save state toggle** — it’s subtle but builds user trust.
- Screenshots were 🔝 — it's awesome to see it all in action.

---

### Re: Your Answers

> Backend data hook-up  
Great to hear you're syncing with Dorian — once it’s live, we’ll be able to do a full test pass across roles. Looking forward to that!

> Avatar upload  
Love the direction — passing from login + editable later = smooth UX. If you need help with preview logic or cloud upload handling, I’m happy to jump in.

> Theme persistence  
No rush for MVP, but agree — storing in `localStorage` will be a great polish point post-demo. Let’s flag it as a possible stretch enhancement if time allows.

---

This is truly a meaningful contribution toward MVP. Clean code, solid design choices, and user-first thinking. Great collaboration overall — excited to test the next wave once backend data flows in! 

Let me know if you want to pair on that or validate anything live.

— *Abdulrahman*

omgdory reviewed (dismissed):
## ✅ Approval Overview
Changes to the frontend look appropriate. Fields representing most data we retain on the backend are in place. However, some fields retained on the backend are missing (see details below). Some superfluous fields, but are good to have, as functionality can be added later down the line if time permits. Dark mode toggle works. Each page of settings makes sense. Code seems clean, but I can't validate for sure as I am unfamiliar with React. Review checklist is in order:
✅ Settings page opens and all panels load correctly
✅ Dark mode applies consistently
✅ Navbar links work for both student and tutor
✅ TutorView uses shared layout and navigation
✅ Code style and routing logic follows project conventions

## 🧠 Details
Fields present on the frontend are missing some of those in the backend, but these fields may be added in a separate PR. For the tutor view, fields for the tutor's subjects as well as their hourly rate are missing. As for the student, there should be fields for grade level, preferred subjects, and emergency contact information (name and email). Again, these fields can be added in a separate PR!

Apart from that, everything else looks good! Integrating the TutorView and StudentView components adds a substantial enhancement to the application's user interface and navigation structure. It provides a degree of separation between how each of these users are meant to interact with our product.

The Profile Settings Page is additionally a good call, as it allows users to manage their account information. The implementation appears to be modular, with the new page encapsulated within its own component, promoting React's strength of reusability and maintainability.

Dark mode integration allows users accustom to dark mode to use LessonConnect without straining their eyes, enhancing accessibility.

Connecting the TutorView and StudentView components with the Navbar further enhances the application's navigation, working to improve the user flow and stopping us from needing to manually type in URL paths.

## 🎯 Next Steps
* ⭐ Connect these settings to backend (update user info, upload avatar, etc.) --> (most important!)
* ⚠️ Add the aforementioned missing fields
* Add error handling and form validation
* Persist theme state across sessions
* Load current user’s data dynamically

aviendha-andrus commented:
This looks fantastic, @FrankFurter123 ! The design is super clean and intuitive. Having a fully structured Profile settings page is such an important part of our MVP. The PR was really well structured and I appreciate the images and clear testing instructions to help visualize and examine the code. I recognize and value all the attention to detail, and I’m excited to see how it continues to evolve when more fields are incorporated like Dorian suggested. Great work!

FrankFurter123 commented:
Hey Dorian! Thank you so much for the thorough review and breakdown — I really appreciate the clear checklist and detailed suggestions!

---
I went ahead and added the **missing fields** that you pointed out:

- For Tutors:
  - `hourlyRate`
  - `subjectsTeach` (courses they teach)
  
  <img width="1296" alt="Screenshot 2025-04-13 at 23 26 38" src="https://github.com/user-attachments/assets/14766d36-1197-4b1f-9220-ab80388a211e" />

-  For Students:
  - `gradeLevel`
  - `preferredSubjects`
  - `EmergencyContact`
  - `EmergencyEmail`

<img width="1411" alt="Screenshot 2025-04-13 at 23 26 55" src="https://github.com/user-attachments/assets/978d29f5-3ef7-481f-b464-9390b1b613a7" />


These are now available in the settings UI, and conditionally rendered based on the role.

>  For now, the way to test this is by **manually switching the user type** between student and tutor. Once the backend is fully integrated, this logic will work automatically during login and route assignment.
<img width="903" alt="Screenshot 2025-04-13 at 23 32 30" src="https://github.com/user-attachments/assets/22d8c9f1-b7d7-4d29-a44f-78cfd03f1dbd" />

---

###  Next Steps

Thanks again for outlining the priorities so clearly — in the next PR, I’ll be focusing on:

- Connecting these fields to the backend to **update and load user info dynamically**
- Handling **avatar uploads**
- Adding **form validation** and proper error handling
- (If time allows) Persisting theme state across sessions using `localStorage`

Thanks again for the detailed feedback and support!

— Frank

AbdulAlharbi reviewed (approved):
✅ **PR Approved – Ready to Merge**

Hey Frank — this is a stellar implementation. You've gone above and beyond by incorporating feedback quickly and cleanly:

- ✅ UX enhancements like bio limits, discard buttons, and success messages are all in.
- ✅ UI polish is top-tier — button labels, visual feedback, and layout all feel production-ready.
- ✅ Your answers to the follow-up questions show strong coordination with backend and forward-thinking design.

Everything works smoothly and aligns with the original issue (#141). This is a **meaningful contribution** that elevates our MVP and sets a strong foundation for user profile management.

Great job!  
— *Abdulrahman*

omgdory reviewed (approved):
Nice work implementing the changes @FrankFurter123! The added fields don't seem to have broken existing code, so looks good!
